cloud/keystone_test.go

```diff
@@ -368,19 +373,52 @@ var _ = Describe("Keystone client", func() {
 			})
 
 			It("Should read tokens scoped to admin project", func() {
-				server.AppendHandlers(
-					ghttp.RespondWithJSONEncoded(201, getV3TokensAdminResponse()),
-					ghttp.RespondWithJSONEncoded(200, getV3TokensAdminResponse()),
-				)
+				setResponses(getV3TokensAdminResponse())
 				setupV3Client()
+
 				auth, err := client.VerifyToken(token)
+
 				Expect(err).To(BeNil())
 				Expect(auth.TenantID()).To(Equal("admin-project-id"))
 				Expect(auth.TenantName()).To(Equal("admin-project"))
 				Expect(auth.DomainID()).To(Equal("default"))
 				Expect(auth.DomainName()).To(Equal("default"))
 				Expect(auth.IsAdmin()).To(BeTrue())
 			})
+
+			It("Should reject admin token with multiple roles", func() {
+				multipleRoles := []interface{}{
+					map[string]interface{}{
+						"id":   "51cc68287d524c759f47c811e6463340",
+						"name": "Member",
+					},
+					map[string]interface{}{
+						"id":   "7f0ea059b6d84029b60c18169d3c1d9a",
+						"name": "admin",
+					},
+				}
+				setResponses(getV3TokensAdminResponseWithRoles(multipleRoles))
+				setupV3Client()
+
+				_, err := client.VerifyToken(token)
```

Comment:
nit: this could be one lined:
```go
Expect(client.VerifyToken(token)).NotTo(Succeed())
```

Reply:
Unfortunately it cannot: `Unexpected non-nil/non-zero extra argument at index 1:`